plugin/al/plugin/AL_USDMayaTestPlugin/AL/usdmaya/commands/test_ProxyShapeSelect.cpp

```diff
@@ -54,25 +93,43 @@ TEST(ProxyShapeSelect, selectNode1)
     };
 
     auto compareNodes = [](const SdfPathVector& paths) {
-        MSelectionList sl;
-        MGlobal::getActiveSelectionList(sl);
-        EXPECT_EQ(sl.length(), paths.size());
-        for (uint32_t i = 0; i < sl.length(); ++i) {
-            MObject obj;
-            sl.getDependNode(i, obj);
-            MFnDependencyNode fn(obj);
-            MStatus           status;
-            MPlug             plug = fn.findPlug("primPath", &status);
-            EXPECT_EQ(MStatus(MS::kSuccess), status);
-            MString pathName = plug.asString();
-            bool    found = false;
-            for (uint32_t j = 0; j < paths.size(); ++j) {
-                if (pathName == paths[j].GetText()) {
-                    found = true;
-                    break;
+        if (!MayaUsdProxyShapePlugin::useVP2_NativeUSD_Rendering()) {
+            MSelectionList sl;
+            MGlobal::getActiveSelectionList(sl);
+            EXPECT_EQ(sl.length(), paths.size());
+
+            for (uint32_t i = 0; i < sl.length(); ++i) {
+                MObject obj;
+                sl.getDependNode(i, obj);
+                MFnDependencyNode fn(obj);
+                MStatus           status;
+                MPlug             plug = fn.findPlug("primPath", &status);
+                EXPECT_EQ(MStatus(MS::kSuccess), status);
+                MString pathName = plug.asString();
+                bool    found = false;
+                for (uint32_t j = 0; j < paths.size(); ++j) {
+                    if (pathName == paths[j].GetText()) {
+                        found = true;
+                        break;
+                    }
                 }
+                EXPECT_TRUE(found);
+            }
+        } else {
+            auto selection = Ufe::GlobalSelection::get();
+            for (auto it : *selection) {
```

Comment:
I realize it's just a test, but it feels like there should be a way to create a UFE path and use Ufe::Selection searching, which is Trie-based and therefore more efficient.

Reply:
Unfortunately not for these specific tests since it's checking that all of the UFE selection items have *some* SdfPath, rather than the other way around. I have however gone ahead and split off the search code into a helper function to clarify that.